test/controllers/stages_controller_test.rb

```diff
@@ -222,6 +233,12 @@ def create_stage(overrides = {})
         end
       end
 
+      it "fails with `no_code_deployed` in params" do
```

Comment:
tests are easier to read/understand if they are less technical and more "user story" in the test cases like `it "fails when trying to set no code deployed"`